src/contracts/core/DelegationManager.sol

```diff
@@ -937,7 +942,24 @@ contract DelegationManager is
             withdrawals[i] = queuedWithdrawals[withdrawalRoots[i]];
             shares[i] = new uint256[](withdrawals[i].strategies.length);
 
-            uint256[] memory slashingFactors = _getSlashingFactors(staker, operator, withdrawals[i].strategies);
+            uint32 slashableUntil = withdrawals[i].startBlock + MIN_WITHDRAWAL_DELAY_BLOCKS;
+
+            uint256[] memory slashingFactors;
+            // if the slashableUntil block is in the future, simply read current slashing factors
+            // still possible however for the slashing factors to change before the withdrawal is completable
+            // and the shares withdrawn to be less
+            if (slashableUntil > uint32(block.number)) {
```

Comment:
this should be >=

alternatively, making it `uint32(block.number) > slashableUntil` and switching the inners of the conditionals is cleaner

Reply:
`_getSlashingFactors` gets the current slashing factor, so the logic here is that if slashableUntil is at some future block, then we simply read the current slashing factors. Else we read it at the exact slashableUntil block

Reply:
Its clearer if we go >=, but the underlying logic is still the same

Reply:
I get now what you guys are saying now and agreed.
Updated now https://github.com/Layr-Labs/eigenlayer-contracts/pull/965/commits/e6c66cd4f62125f9b5058dcf59ea686d279ccc75